### ISSUE TITLE
`bacalhau` fails to `get`

### ISSUE THREAD

**Issue by dtbuchholz** [bacalhau-project/bacalhau#3003]
## Summary

Using `bacalhau` will successfully start up, `run` jobs, and `describe` them. However, the `get` command hangs and exits early. This is also true when doing [local dev](https://github.com/bacalhau-project/bacalhau/blob/main/docs/running_locally.md) with `bacalhau devstack`

## Details
You can test this in either production or dev env:
1. Set up the dev env: `bacalhau devstack`
2. Copy the env vars in a separate window:
   ```
   export BACALHAU_NODE_IPFS_SWARMADDRESSES=/ip4/127.0.0.1/tcp/58669/p2p/QmPPGBfhzWYfA6hBhkfskoxNQMhgDUDwCt8y4s8E6mT39d
   export BACALHAU_NODE_CLIENTAPI_HOST=0.0.0.0
   export BACALHAU_NODE_CLIENTAPI_PORT=58658
   export BACALHAU_NODE_LIBP2P_PEERCONNECT=/ip4/127.0.0.1/tcp/58651/p2p/QmPZUMAf6VkygJZ6C4JaxonNXFFUJKLp2LHQ7t1ojm9Jag,/ip4/127.0.0.1/tcp/58661/p2p/QmUfykjRmiQhMUZdAXva2rKa5jySnc5g3BWF1xWrywS4J7,/ip4/127.0.0.1/tcp/58666/p2p/QmdEbzT6BvqMRXDn1spTU5v3wmSNo1JUfHq3XwYPtdqJ66,/ip4/127.0.0.1/tcp/58671/p2p/QmRtpY9DA8HHQA8riM7ifFf5yk73NY2wmPpwRUcBftJQ29
   ```
3. Run a simple job (success): `bacalhau docker run ubuntu echo "hello devstack test"`
    <details>
    <summary>output</summary>
    <pre>
    Using default tag: latest. Please specify a tag/digest for better reproducibility.
    Job successfully submitted. Job ID: a9f4255a-075d-44a0-ba3b-445c73987fc6
    Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):
    
	    Communicating with the network  ................  done ✅  0.0s
	       Creating job for submission  ................  done ✅  0.5s
	                   Job in progress  ................  done ✅  0.0s
    
    To download the results, execute:
	    bacalhau get a9f4255a-075d-44a0-ba3b-445c73987fc6
    
    To get more details about the run, execute:
	    bacalhau describe a9f4255a-075d-44a0-ba3b-445c73987fc6
    </pre>
    </details>
4. Describe it (success): `bacalhau describe a9f4255a-075d-44a0-ba3b-445c73987fc6`
    <details>
    <summary>output</summary>
    <pre>
      Job:
    APIVersion: V1beta2
    Metadata:
      ClientID: d579c4b45f4d4a3520f9137ae5d7c672e6940a248d98e5dc33e85d2600deef29
      CreatedAt: "2023-11-10T10:02:21.552612-08:00"
      ID: a9f4255a-075d-44a0-ba3b-445c73987fc6
      Requester:
        RequesterNodeID: QmPZUMAf6VkygJZ6C4JaxonNXFFUJKLp2LHQ7t1ojm9Jag
        RequesterPublicKey: CAASpgIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCnJfdgEbZibQcMLT6kwQgzqQA6Pck9nE12+XCm9En5qbhERTXwkavC+NWcobD10FriEUPGUsnZcK/V8+kl0A/2VV4IZqgS1o7n5tUMMgf2EMoXO3G+aGS7g9Q7LqLrwSd/3iXMLn24rXiFRvXs5wilV922OUL+YSiDoeXuP2cItlrnJ5WxW6K6vtMJ/tLoqdpL66omkLV6uMxj+19vGrHwYVsAFaiWilz3HZzqPWKBzBBkiu57oipsMtpVORQIqeWknfHdFnUPLxzcAhKVsi7ehMm6lJ5WDoNj2G+0T5l0/K2uzNWeGXRRWxEcEORJTvME93oKJzmsg7rLZO3HbyAdAgMBAAE=
    Spec:
      Deal:
        Concurrency: 1
      Docker: {}
      EngineSpec:
        Params:
          Entrypoint: null
          EnvironmentVariables: []
          Image: ubuntu
          Parameters:
          - echo
          - hello devstack test
          WorkingDirectory: ""
        Type: docker
      Network:
        Type: None
      Outputs:
      - Name: outputs
        Path: /outputs
      PublisherSpec:
        Type: ipfs
      Resources:
        GPU: ""
      Timeout: 1800
      Wasm:
        EntryModule: {}
    State:
      CreateTime: "2023-11-10T10:02:21.552612-08:00"
      Executions:
      - ComputeReference: e-2a966e24-fdec-407e-b97c-7331b0691df0
        CreateTime: "2023-11-10T10:02:21.572682-08:00"
        DesiredState: 2
        JobID: a9f4255a-075d-44a0-ba3b-445c73987fc6
        NodeId: QmdEbzT6BvqMRXDn1spTU5v3wmSNo1JUfHq3XwYPtdqJ66
        PublishedResults:
          CID: QmarGkxwnx2jXiJj2iUS9oJuPBGWUxLRpuizo92wz5Ue7r
          StorageSource: ipfs
        RunOutput:
          exitCode: 0
          runnerError: ""
          stderr: ""
          stderrtruncated: false
          stdout: |
            hello devstack test
          stdouttruncated: false
        State: Completed
        Status: . execution completed
        UpdateTime: "2023-11-10T10:02:21.918671-08:00"
        Version: 6
      JobID: a9f4255a-075d-44a0-ba3b-445c73987fc6
      State: Completed
      TimeoutAt: "2023-11-10T10:32:21.552612-08:00"
      UpdateTime: "2023-11-10T10:02:21.936141-08:00"
      Version: 3
      </pre>
      </details>
5. Get data (fails): `bacalhau get a9f4255a-075d-44a0-ba3b-445c73987fc6`
    - `bacalhau list` showed the job as "Completed"
    - You'll notice the published result above can be fetched manually (e.g., [in browser](ipfs://bafybeifz4lmp53z5dhnwdfcgettzdg7sgiprw3fsksql2nuyrxsumlvk3e/)) at `QmarGkxwnx2jXiJj2iUS9oJuPBGWUxLRpuizo92wz5Ue7r`, but the `get` command fails to return data.
    <details>
    <summary>output</summary>
    <pre>
    Fetching results of job 'a9f4255a-075d-44a0-ba3b-445c73987fc6'...
    error downloading job: failed to get ipfs cid 'QmarGkxwnx2jXiJj2iUS9oJuPBGWUxLRpuizo92wz5Ue7r': context deadline exceeded
    </pre>
    </details>
    <details>
    <summary>details from `LOG_LEVEL=debug` where the pattern shown repeats indefinitely</summary>
    <pre>
    10:10:07.933 | INF pkg/publicapi/middleware/logger.go:42 > [ClientID:] [Duration:0.244167] [JobID:] [Method:GET] [NodeID:QmPZUMAf] [Referer:] [RemoteAddr:127.0.0.1] [RequestID:Bhx1XNaZfnctxZEZMwXA7qfoe70bKA6R] [Size:111] [StatusCode:200] [URI:/api/v1/orchestrator/jobs/a9f4255a-075d-44a0-ba3b-445c73987fc6/results] [UserAgent:Go-http-client/1.1]
    10:10:12.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:10:13.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:10:13.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    10:10:22.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:10:23.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:10:23.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    10:10:32.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:10:33.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:10:33.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    10:10:42.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:10:43.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:10:43.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    10:10:52.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:10:53.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:10:53.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    10:11:02.768 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmUfykjR]
    10:11:03.184 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmdEbzT6]
    10:11:03.364 | DBG pkg/compute/sensors/logging_sensor.go:50 > ActiveJobs: [] [NodeID:QmRtpY9D]
    </pre>
    </details>

### Environment
- MacOS M1, Ventura 13.4.1
- Go (not sure if relevant): go1.21.3 darwin/arm64
- Note: I couldn't [build from source](https://github.com/bacalhau-project/bacalhau/blob/main/docs/running_locally.md#optional-building-bacalhau-from-source) due to a go1.20 requirement, so I used the standard setup with `curl -sL https://get.bacalhau.org/install.sh | bash`
- bacalhau version info:
   - client: v1.1.4
   - server:  v1.1.4
   - latest:  v1.1.4

**Comment by rossjones** [comment]
I had the `get` call work once out of 10 attempts, so there's definitely something strange going on.
I checked out commit 03bba26 (v1.1.3) and it worked fine, so it looks like the bug may have been introduced in 1.1.4

**Comment by aronchick** [comment]
Amazing debugging and bug report! Thank you!

**Comment by dtbuchholz** [comment]
Quick question—how to I force the server to be the same version as the client? E.g., if I build from source or [download the tarball](https://github.com/bacalhau-project/bacalhau/releases/tag/v1.1.3) for `v1.1.3` and then run `bacalhau version`, it'll log the client as `v1.1.3` but server as `v1.1.4`. It appears the issue is on the server side because `get` still hangs with the client as `v1.1.3`.

**Comment by wdbaruni** [comment]
I believe this is related to changing the IPFS swarm addrs in our network. To verify this is root cause, provide the new addrs to your command using:
```
bacalhau get 905f4af4-e88e-4bf8-af8f-820314de56f3 --ipfs-swarm-addrs /ip4/35.245.161.250/tcp/4001/p2p/12D3KooWAQpZzf3qiNxpwizXeArGjft98ZBoMNgVNNpoWtKAvtYH,/ip4/34.86.254.26/tcp/4001/p2p/12D3KooWLfFBjDo8dFe1Q4kSm8inKjPeHzmLBkQ1QAjTHocAUazK,/ip4/35.245.215.155/tcp/4001/p2p/12D3KooWH3rxmhLUrpzg81KAwUuXXuqeGt4qyWRniunb5ipjemFF,/ip4/34.145.201.224/tcp/4001/p2p/12D3KooWBCBZnXnNbjxqqxu2oygPdLGseEbfMbFhrkDTRjUNnZYf,/ip4/34.145.201.224/udp/4001/quic/p2p/12D3KooWBCBZnXnNbjxqqxu2oygPdLGseEbfMbFhrkDTRjUNnZYf,/ip4/35.245.41.51/tcp/4001/p2p/12D3KooWJM8j97yoDTb7B9xV1WpBXakT4Zof3aMgFuSQQH56rCXa,/ip4/35.245.41.51/udp/4001/quic/p2p/12D3KooWJM8j97yoDTb7B9xV1WpBXakT4Zof3aMgFuSQQH56rCXa 
```

If this works, and to avoid passing `--ipfs-swarm-addrs` each time, you need to:
1. Upgrade your cli to `v1.1.4` as it has to correct defaults
2. Remove or update Bacalhau configs, which are usually under `~/.bacalhau/config.yaml`. If you didn't modify the configs before, then deleting the file should be enough as the defaults will be picked up. Otherwise update or delete `Node.IPFS.SwarmAddresses` entry

**Comment by wdbaruni** [comment]
Tracking the fix here
https://github.com/bacalhau-project/bacalhau/issues/3004

**Comment by rossjones** [comment]
This does also happen with devstack, might it be outputting the wrong addresses?

**Comment by rossjones** [comment]
I believe the fix for #3004 is in and so we're safe to close this issue as it is definitely working with 1.1.5
